src/Connection.ts

```diff
@@ -13,13 +14,18 @@ export class Socks5Connection {
     private errorHandler = () => {};
     public metadata: any = {};
 
-    constructor(server: Socks5Server, socket: Duplex) {
+    constructor(server: Socks5Server, socket: net.Socket) {
         this.socket = socket;
         this.server = server;
 
-        socket.on('error', this.errorHandler); // Ignore errors
+        if (socket.remoteAddress) {
+            this.destAddress = net.isIPv4(socket.remoteAddress) ? socket.remoteAddress : socket.remoteAddress.replace(/^.*:/, '');
```

Comment:
destAddress and destPort are used for the target destination that the socks5 client is trying to initialize a tunnel to, I think it would be confusing to use these for the client's IP address and then switch it to the target destination once the connection request is received